testing/test_mark.py

```diff
@@ -973,6 +976,53 @@ def test_addmarker_order():
     extracted = [x.name for x in node.iter_markers()]
     assert extracted == ["baz", "foo", "bar"]
 
+    # Check marks/keywords with Function.
+    session.name = "session"
+    session.keywords = NodeKeywords(session)
+
+    # Register markers for `--strict-markers`.
+    added_markers = pytestconfig._inicache["markers"] + [
+        "funcmark",
+        "prepended",
+        "funcmark2",
+    ]
+    monkeypatch.setitem(pytestconfig._inicache, "markers", added_markers)
+
+    @pytest.mark.funcmark
+    def f1():
+        assert False, "don't call me"
+
+    func = Function.from_parent(node, name="func", callobj=f1)
+    expected_marks = ["funcmark", "baz", "foo", "bar"]
+    assert [x.name for x in func.iter_markers()] == expected_marks
+    func.add_marker("prepended", append=False)
+    assert [x.name for x in func.iter_markers()] == ["prepended"] + expected_marks
+    assert set(func.keywords) == {
+        "Test",
+        "bar",
+        "baz",
+        "foo",
+        "func",
+        "funcmark",
+        "prepended",
+        "pytestmark",
+        "session",
+    }
+
+    # Changing the "obj" updates marks and keywords (lazily).
```

Comment:
"keywords" being updated is not done here yet (but in https://github.com/blueyed/pytest/pull/359 already).